Sample code to accept JSON POST via HTTP and advertise as a beacon

mharper opened the pull request:
The purpose of this is:
1) To satisfy a request by @greencoder to be able to advertise as a beacon by performing a `POST` from Javascript.
2) To investigate how a scanner hub could be directed remotely to advertise as a beacon.

This represents my _second_ Go program, so I expect there's a ton of stylistic and idiomatic changes that should be made here. Let me have it @syoder!

mharper committed:
Sample code to accept JSON POST via HTTP and advertise as a beacon

mharper commented:
I was just going full-steam ahead here in order to see if I could browbeat the existing code into working via HTTP POST.  I think it's now time to be a better citizen and actually get tests into the loop.  And @syoder I'm pretty sure I'm doing a bunch of first-letter capitalization that's wrong, correct?

mharper committed:
Removes commented-out code.

mharper committed:
Changes `Beacon` to `BeaconSpecification`.

Seems a little more specific.  (See what I did there?)

Struct and its fields must be exported in order for JSON Decoder to
work.

http://stackoverflow.com/a/34696916/128317

syoder commented on altbeacon.go:
```diff
@@ -0,0 +1,17 @@
+	beaconIds := AltBeaconFields(uuid, major, minor)
+	beacon := NewBeacon("altbeacon",
+		beaconIds,                   // ids
+		Fields{FieldFromInt8(0x20)}, // data
```
Maybe this data field should be a parameter that gets passed in?

mharper replied:
Absolutely. That was a brute force hack just to see if this would work at all.

mharper committed:
Moves `http_beacon.go` into its own package `http_beacon`

This makes it considerably easier to generate a separate executable for
`http_beacon`, including the ability to cross-compile for running on a pi.

mharper committed:
Adds documentation for `http_beacon`

Includes format of JSON to POST as well as directions on
cross-compiling.
